src/diffusers/loaders.py

```diff
@@ -2042,24 +2077,42 @@ def fuse_lora(self, fuse_unet: bool = True, fuse_text_encoder: bool = True, lora
         if fuse_unet:
             self.unet.fuse_lora(lora_scale)
 
-        def fuse_text_encoder_lora(text_encoder):
-            for _, attn_module in text_encoder_attn_modules(text_encoder):
-                if isinstance(attn_module.q_proj, PatchedLoraProjection):
-                    attn_module.q_proj._fuse_lora(lora_scale)
-                    attn_module.k_proj._fuse_lora(lora_scale)
-                    attn_module.v_proj._fuse_lora(lora_scale)
-                    attn_module.out_proj._fuse_lora(lora_scale)
+        if self.use_peft_backend:
+            from peft.tuners.tuners_utils import BaseTunerLayer
+
+            def fuse_text_encoder_lora(text_encoder, lora_scale=1.0):
+                for module in text_encoder.modules():
+                    if isinstance(module, BaseTunerLayer):
+                        if lora_scale != 1.0:
+                            module.scale_layer(lora_scale)
 
-            for _, mlp_module in text_encoder_mlp_modules(text_encoder):
-                if isinstance(mlp_module.fc1, PatchedLoraProjection):
-                    mlp_module.fc1._fuse_lora(lora_scale)
-                    mlp_module.fc2._fuse_lora(lora_scale)
+                        module.merge()
+
+        else:
+            warnings.warn(
+                "You are using an old version of LoRA backend. This will be deprecated in the next releases in favor of PEFT"
+                " make sure to install the latest PEFT and transformers packages in the future.",
+                FutureWarning,
+            )
```

Comment:
```suggestion
            deprecate
```

Can you please use the deprecate function here? For an example, see: https://github.com/huggingface/diffusers/blob/e312b2302b5445271198ebed8f2fbcd543633f78/src/diffusers/loaders.py#L2121